File: Leetcode July Challenge/Python/Valid Triangle Number.py
class Solution:
    def triangleNumber(self, nums: List[int]) -> int:
        n = len(nums)
        nums.sort()
        ans = 0
        for i in range(n-1, 1, -1):
            lo = 0
            hi = i-1
            while lo < hi:
                if (nums[lo] + nums[hi]) > nums[i]:
                    ans += (hi - lo)
                    hi -= 1
                else:
                    lo += 1
        return ans            
                    
        

File: Leetcode July Challenge/C++/Valid Triangle Number.cpp
class Solution {
public:
    int triangleNumber(vector<int>& nums) {
        
        int ans = 0, n = nums.size();
        sort(nums.begin(), nums.end());
        
        for(int i = n - 1; i >= 2; i--){
            
            int lo = 0, hi = i - 1;
            while(lo < hi){
                if(nums[lo] + nums[hi] > nums[i]){
                    ans += (hi - lo);
                    hi--;
                }else{
                    lo++;
                }
            }
            
        }
        
        return ans;
    }
};
